Fork_Maintenance.md

```diff
@@ -0,0 +1,263 @@
+# OpenHands Fork Maintenance Guide
+
+This guide outlines a strategy for maintaining a fork of the main OpenHands repository while incorporating custom patches and ensuring stability and security.
+
+## Key Goals
+
+1.  **Consistent Container Images:** Use a specific, matched pair of `runtime` and `agent` images across all environments.
+2.  **Security & Intentional Updates:** Avoid automatic updates from upstream; intentionally review and integrate changes.
+3.  **Local Patch Management:** Easily add, manage, and remove custom features or external PRs.
+
+## 1. Initial Setup
+
+First, ensure your fork is cloned locally and the original OpenHands repository is added as a remote named `upstream`.
+
+```bash
+# Clone your fork (replace with your fork URL)
+# git clone <your-fork-url>
+# cd OpenHands
+
+# Add the main OpenHands repository as the 'upstream' remote
+git remote add upstream https://github.com/All-Hands-AI/OpenHands.git
+
+# Fetch upstream branches and tags
+git fetch upstream
+```
+
+## 2. Container Image Management
+
+To ensure consistency, build, tag, and distribute your own `runtime` and `agent` images based on your primary integration branch.
+
+**a. Building Images:**
+
+*   **Source Branch:** Images should be built by CI/CD processes triggered by commits to the `release/stable-with-patches` branch.
```

Comment:
Where do we want to build and host these images? These will be used by github actions and local running of openhands.

We currently host packages in dockerhub, ECR, and github packages (generally not images, but gems and stuff).

I like github packages for development tools, but we could double down on ECR if that keeps things simpler.

Should these images be public or private? At the moment this is a public fork. There's nothing specific to our code or architecture here, just a couple of neat features we would use that may or may not be on their way into upstream.